official/resnet/resnet.py

```diff
@@ -608,3 +676,13 @@ def __init__(self, resnet_size_choices=None):
              'is not always compatible with CPU. If left unspecified, '
              'the data format will be chosen automatically based on '
              'whether TensorFlow was built for CPU or GPU.')
+
+    self.add_argument(
+        '--multi_gpu', action='store_true',
+        help='If set, run across all available GPUs. Note that this is '
+        'superseded by the --num_gpus flag.')
+
+    self.add_argument(
+        '--num_gpus', type=int, default=None,
```

Comment:
Why have two flags, seems we could do with just one?

Reply:
CC @tfboyd who requested the second one. In Toby's version, there num_gpus takes many values representing CPU, num gpu, and all GPU, but I found the multi-state indicator confusing. I thought it better to have a simple flag that people can use to mean "use all," and a expert-use-case flag for specifying the exact number. @tfboyd , thoughts on this? Especially versus using CUDA_VISIBLE_DEVICES?

Reply:
Yup, CUDA_VISIBLE_DEVICES seems reasonable to me, and then just a single `--multi-gpu=<boolean>` flag.

Reply:
There was some reason @tfboyd was mildly opposed to that solution... @tfboyd , thoughts?

Reply:
Removed for now as it's not going to be in the final version of replication mode as things currently stand anyways. We can reassess if that causes undue pain for you, @tfboyd